ocaml/xapi/xapi_vm_group_helpers.ml

```diff
@@ -0,0 +1,173 @@
+(*
+ * Copyright (C) 2024 Cloud Software Group
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published
+ * by the Free Software Foundation; version 2.1 only. with the special
+ * exception on linking described in file LICENSE.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *)
+
+module D = Debug.Make (struct let name = "xapi_vm_group_helpers" end)
+
+open D
+
+let check_breach ~__context ~group =
+  let vms =
+    Db.VM_group.get_VMs ~__context ~self:group
+    |> List.filter (fun vm ->
+           Db.VM.get_power_state ~__context ~self:vm = `Running
+       )
+  in
+  let get_host vm =
+    let host =
+      match Db.VM.get_resident_on ~__context ~self:vm with
+      | h when h <> Ref.null ->
+          Some h
+      | _ ->
+          None
+    in
+    host
+  in
+  match vms with
+  | [] | [_] ->
+      None
+  | vm :: _ ->
```

Comment:
When there is only 1 running VMs, not a breach.

Reply:
When there is only 1 running VM, it will meet the conditions of the first case and return None, indicating no breach. I believe the current logic adequately addresses this scenario